src/adapter/search-response-adapter/pagination-adapter.ts

```diff
@@ -1,21 +1,55 @@
-/**
- * Slice the requested hits based on the pagination position.
- *
- * @param  {Record<string} hits
- * @param  {number} page
- * @param  {number} hitsPerPage
- * @returns {Array}
- */
-export function adaptPagination(
-  hits: Record<string, any>,
-  page: number,
+import type {
+  MeiliSearchResponse,
+  PaginationState,
+  InstantSearchPagination,
+} from '../../types'
+import { ceiledDivision, floorDivision } from '../../utils'
+
+function adaptNbPages(
+  searchResponse: MeiliSearchResponse<Record<string, any>>,
   hitsPerPage: number
-): Array<Record<string, any>> {
-  if (hitsPerPage < 0) {
-    throw new TypeError(
-      'Value too small for "hitsPerPage" parameter, expected integer between 0 and 9223372036854775807'
-    )
+): number {
+  if (searchResponse.totalPages != null) {
```

Comment:
Some places you opted by `undefined` and others like this `null`, is there any particular reason for that?

Reply:
I dont see a `!= undefined` in my code. Could you show an example?

Reply:
Absolutely! Thanks for the catch